Rocket.Chat/API/Clients/UploadClient.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  UploadClient.swift
+//  Rocket.Chat
+//
+//  Created by Matheus Cardoso on 12/14/17.
+//  Copyright © 2017 Rocket.Chat. All rights reserved.
+//
+
+import Foundation
+
+struct UploadClient: APIClient {
+    let api: AnyAPIFetcher
+    init(api: AnyAPIFetcher) {
+        self.api = api
+    }
+
+    func upload(roomId: String, data: Data, filename: String, mimetype: String, completion: (() -> Void)? = nil, versionFallback: (() -> Void)? = nil) {
+        let req = UploadRequest(
+            roomId: roomId,
+            data: data,
+            filename: filename,
+            mimetype: mimetype
+        )
+
+        api.fetch(req, succeeded: { result in
+            if let error = result.error {
+                Alert(key: "alert.upload_error").withMessage(error).present()
```

Comment:
status code 2xx with error (I know it's API's fault)? 😂

Reply:
🤷‍♀️